### ISSUE TITLE
CI with tests

### ISSUE THREAD

**Issue by viovanov** [SUSE/scf#2725]

**Comment by gaktive** [comment]
Does this differ from https://github.com/SUSE/scf/pull/2712 which was merged?

**Comment by mudler** [comment]
I'm working on it - there is already a pipeline running Eirini tests against develop overnight, and there is also a pipeline for v3.

It deploys successfully, but it can't login to the cluster. It just needs tweaking, will keep this Issue updated.

**Comment by viovanov** [comment]
@f0rmiga there's some miscommunication here - looks like you've both been working on pipelines.
We've discussed this in the past, but let's keep the conversation here from now on.

@mudler the Concourse pipeline you're building is behind a firewall, so not as useful as the public GitLab one - but I don't mind keeping both.

I'd like to understand the different approaches between these 2
- how much has been implemented (build, test, publish)
- what Kubernetes is being used?

**Comment by mudler** [comment]
> @f0rmiga there's some miscommunication here - looks like you've both been working on pipelines.
> We've discussed this in the past, but let's keep the conversation here from now on.
> 
> @mudler the Concourse pipeline you're building is behind a firewall, so not as useful as the public GitLab one - but I don't mind keeping both.

The pipeline is using https://github.com/SUSE/catapult and EKCP to provide ephemeral cluster dynamically, so each deployment is (almost) reproducible also locally with kind. So as long as the gitlab workers can contact a network behind fw that provides clusters we are good.

E.g. with Catapult, to get a v3 deployment locally now it should be sufficient to (locally):

     $> git clone https://github.com/SUSE/catapult && cd catapult
     $> SCF_OPERATOR=true CHART_URL="https://scf-v3.s3.amazonaws.com/scf-3.0.0-0074f73.tgz" make all
     $> source buildkind/.envrc
     $> cf apps ...

You would ask why, but on the same way you can just do:

     $> git clone https://github.com/SUSE/catapult && cd catapult
     $> SCF_OPERATOR=true CHART_URL="https://scf-v3.s3.amazonaws.com/scf-3.0.0-0074f73.tgz" make all smoke terminal
     $> source buildkind/.envrc
     $> cf apps ...

or `make all cats` https://github.com/SUSE/catapult/wiki/Running-SCF-tests

It gets a pristine environment each time and tests are being run afterwards automatically.

> 
> I'd like to understand the different approaches between these 2
> 
>     * how much has been implemented (build, test, publish)

It currently deploys SCFv3 from the latest develop charts published in the `scf-v3` S3 bucket and runs smoke tests right after. The pipelines now are currently failing (I suspect) for https://github.com/SUSE/scf/issues/2867 ( https://concourse.suse.de/teams/main/pipelines/eirini-ci/jobs/smokes-eirini-v3-develop/builds/46 )

> 
>     * what Kubernetes is being used?

It uses Kind (2.1, for compatibility with SCFv2 as public charts on crio doesn't deploy yet)

**Comment by mudler** [comment]
Update: The same smoke tests with diego on v3 are passing https://concourse.suse.de/teams/main/pipelines/eirini-ci/jobs/smokes-diego-v3-develop/builds/1 . So it seems indeed the pipeline runs tests correctly now

**Comment by f0rmiga** [comment]
@viovanov I have not implemented any of the tests on Gitlab. It's only doing linting, building and publishing. Right now, we are using the shared Gitlab runners for this process. We can at any moment add runners that are behind the VPN as they act as clients connecting to the Gitlab server. This would solve both the issue of making the pipelines publicly visible and also being able to use the ephemeral k8s clusters that @mudler is using. This was my understanding when we discussed the use of Gitlab - that we would start using it as the main CI for v3 moving forward.

**Comment by viovanov** [comment]
@f0rmiga yes, that's fine - I thought you had already started work on running tests.
So if I understand correctly, you can re-use @mudler's work to run smoke tests from a gitlab runner? (and it would show up in the public pipeline)?

**Comment by f0rmiga** [comment]
@viovanov I believe so. We will need to spawn a few runners on the internal cluster first to try it out.
